autodistill/cli.py

```diff
@@ -0,0 +1,167 @@
+import json
+import os
+
+import click
+import cv2
+import roboflow
+import supervision as sv
+
+from autodistill.detection.caption_ontology import CaptionOntology
+from autodistill.registry import import_requisite_module
+
+SUPPORTED_ROBOFLOW_MODEL_UPLOADS = ["yolov5", "yolov5-seg", "yolov8", "yolov8-seg"]
+
+SUPPORTED_MODEL_TYPES = ["detection", "segmentation" "classification"]
+
+SUPPORTED_DATASET_FORMATS = ["yolov8", "yolov5", "voc"]
+
+
+@click.command()
+@click.argument("images")
+@click.option("--base", default="grounding_dino")
+@click.option("--target", default="yolov8")
+@click.option("--model_type", default="detection")
+@click.option("--ontology", default={}, required=True)
+@click.option("--epochs", default=200, required=True)
+@click.option("--output", default="./dataset", required=True)
+@click.option("--upload-to-roboflow", default=False, required=False)
+@click.option("--project_license", default="MIT", required=False)
+@click.option("--dataset_format", default="voc", required=False)
+def main(
+    images, base, target, model_type, ontology, epochs, output, upload_to_roboflow, project_license, dataset_format
+):
+    if dataset_format not in SUPPORTED_DATASET_FORMATS:
+        print(
+            "Dataset format not supported. Please choose from the following dataset formats: "
+            + str(SUPPORTED_DATASET_FORMATS)
+        )
+        exit()
+
+    if model_type not in SUPPORTED_MODEL_TYPES:
+        print(
+            "Model type not supported. Please choose from the following model types: "
+            + str(SUPPORTED_MODEL_TYPES)
+        )
+        exit()
+
+    if ontology == "{}":
+        print("No ontology provided. Please provide an ontology.")
+        exit()
+
+    print("Loading base model...")
+    model = import_requisite_module(base)
+
+    try:
+        ontology = json.loads(ontology, strict=False)
+    except:
+        print("Ontologies must be valid JSON.")
+        exit()
+        
+    if upload_to_roboflow:
+        roboflow.login()
+
+        if model_value not in SUPPORTED_ROBOFLOW_MODEL_UPLOADS:
+            print(
+                f"Model type {model_value} is not supported by Roboflow. Please choose one of {SUPPORTED_ROBOFLOW_MODEL_UPLOADS}."
```

Comment:
Explicitly call out for deployment so it's clear that we're not saying Roboflow can't work with it at all.